bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/chart/defaultchartprovider/DefaultChartProvider.java

```diff
@@ -141,13 +134,25 @@ public String getName() {
     }
 
     @Override
-    public BufferedImage createChart(String service, String theme, Date startTime, Date endTime, int height, int width,
-            String items, String groups, Integer dpiValue, Boolean legend)
+    public BufferedImage createChart(String serviceId, String theme, Date startTime, Date endTime, int height,
+            int width, String items, String groups, Integer dpiValue, Boolean legend)
             throws ItemNotFoundException, IllegalArgumentException {
         logger.debug(
                 "Rendering chart: service: '{}', theme: '{}', startTime: '{}', endTime: '{}', width: '{}', height: '{}', items: '{}', groups: '{}', dpi: '{}', legend: '{}'",
-                service, theme, startTime, endTime, width, height, items, groups, dpiValue, legend);
-        QueryablePersistenceService persistenceService;
+                serviceId, theme, startTime, endTime, width, height, items, groups, dpiValue, legend);
+
+        // If a persistence service is specified, find the provider, or use the default provider
+        PersistenceService service = (serviceId == null) ? persistenceServiceRegistry.getDefault()
+                : persistenceServiceRegistry.get(serviceId);
+
+        // Did we find a service?
+        QueryablePersistenceService persistenceService = (service instanceof QueryablePersistenceService)
+                ? (QueryablePersistenceService) service
+                : (QueryablePersistenceService) persistenceServiceRegistry.getAll() //
+                        .stream() //
+                        .filter(it -> it instanceof QueryablePersistenceService) //
+                        .findFirst() //
+                        .orElseThrow(() -> new IllegalArgumentException("No Persistence service found."));
```

Comment:
FYI: To filter a stream for specific instances a combination of filter and map could be useful:

```Java
stream().
  filter(QueryablePersistenceService.class::isInstance)
  .map(QueryablePersistenceService.class::cast)
  .findFirst()
```